benchmarks/scripts/benchmarks.py

```diff
@@ -0,0 +1,316 @@
+            if os.path.exists(script_file_name):
+                os.remove(script_file_name)
+        print(">>> Benchmark script generated and uploaded\n")
+
+        # Start the script
+        job_cmd = (
+            f"ssh -i {ssh_id_file} hadoop@{cluster_hostname} " +
+            f"screen -d -m bash {script_file_name}"
+        )
+        print(job_cmd)
+        run_cmd(job_cmd, stream_output=True)
+
+        # Print the screen where it is running
+        run_cmd(f"ssh -i {ssh_id_file} hadoop@{cluster_hostname}" +
+                f""" "screen -ls ; sleep 2; echo Files for this benchmark: ; ls {self.benchmark_id}*" """,
+                stream_output=True, throw_on_error=False)
+        print(f">>> Benchmark id {self.benchmark_id} started in a screen. Stdout piped into {self.output_file}. "
+              f"Final report will be generated on completion in {self.json_report_file}.\n")
+
+    @staticmethod
+    def output_file(benchmark_id):
+        return f"{benchmark_id}-out.txt"
+
+    @staticmethod
+    def json_report_file(benchmark_id):
+        return f"{benchmark_id}-report.json"
+
+    @staticmethod
+    def csv_report_file(benchmark_id):
+        return f"{benchmark_id}-report.csv"
+
+    @staticmethod
+    def completed_file(benchmark_id):
+        return f"{benchmark_id}-completed.txt"
+
+    @staticmethod
+    def wait_for_completion(cluster_hostname, ssh_id_file, benchmark_id, copy_report=True):
+        completed = False
+        succeeded = False
+        output_file = Benchmark.output_file(benchmark_id)
+        completed_file = Benchmark.completed_file(benchmark_id)
+        json_report_file = Benchmark.json_report_file(benchmark_id)
+        csv_report_file = Benchmark.csv_report_file(benchmark_id)
+
+        print(f"\nWaiting for completion of benchmark id {benchmark_id}")
+        while not completed:
+            # Print the size of the output file to show progress
+            (_, out, _) = run_cmd_over_ssh(f"stat -c '%n:   [%y]   [%s bytes]' {output_file}",
+                                           cluster_hostname, ssh_id_file,
+                                           throw_on_error=False)
+            out = out.decode("utf-8").strip()
+            print(out)
+            if "No such file" in out:
+                print(">>> Benchmark failed to start")
+                return
+
+            # Check for the existence of the completed file
+            (_, out, _) = run_cmd_over_ssh(f"ls {completed_file}", cluster_hostname, ssh_id_file,
+                                           throw_on_error=False)
+            if completed_file in out.decode("utf-8"):
+                completed = True
+            else:
+                time.sleep(60)
+
+        # Check the last few lines of output files to identify success
+        (_, out, _) = run_cmd_over_ssh(f"tail {output_file}", cluster_hostname, ssh_id_file,
+                                       throw_on_error=False)
+        if "SUCCESS" in out.decode("utf-8"):
```

Comment:
The JSON report is printed after `SUCCESS` so successful runs are being reported as a failure here.

Reply:
Aargh. fixed it. thanks for testing this out!!